vessim/sil.py

```diff
@@ -141,32 +162,37 @@ def start(self, microgrid: Microgrid) -> None:
 
         Thread(target=self._collect_set_requests_loop, daemon=True).start()
 
-    def step(self, time: datetime, p_delta: float, actor_infos: dict) -> None:
+    def step(self, time: datetime, p_delta: float, e_delta: float, actor_infos: dict) -> None:
         assert self.microgrid is not None
-        self.broker._add_microgrid_data(
-            time,
-            {
-                "microgrid": self.microgrid,
-                "actor_infos": actor_infos,
-                "p_delta": p_delta,
-            },
+        self.data_pipe_in.send(
+            (
+                time,
+                {
+                    "microgrid": self.microgrid,
+                    "actor_infos": actor_infos,
+                    "p_delta": p_delta,
+                    "e_delta": e_delta,
+                },
+            )
         )
 
-    def finalize(self) -> None:
-        self.broker._finalize()
-
     def _collect_set_requests_loop(self):
         while True:
+            start_time = time.time()
             events_by_category = defaultdict(dict)
-            for event in self.broker._consume_events():
+            while self.events_pipe_out.poll():
+                event = self.events_pipe_out.recv()
                 events_by_category[event["category"]][event["time"]] = event["value"]
             for category, events in events_by_category.items():
                 self.request_collectors[category](
-                    events=events_by_category[category],
+                    events=events,
                     microgrid=self.microgrid,
-                    kwargs=self.kwargs,
                 )
-            sleep(self.request_collector_interval)
+            # Calculate elapsed time and sleep if necessary
+            elapsed_time = time.time() - start_time
```

Comment:
```suggestion
            elapsed_time = time.monotonic() - start_time
```
(See suggestion above.)